zerver/views/realm.py

```diff
@@ -327,11 +327,15 @@ def check_subdomain_available(request: HttpRequest, subdomain: str) -> HttpRespo
 
 def realm_reactivation(request: HttpRequest, confirmation_key: str) -> HttpResponse:
     try:
-        realm = get_object_from_key(confirmation_key, [Confirmation.REALM_REACTIVATION])
+        realm = get_object_from_key(
+            confirmation_key, [Confirmation.REALM_REACTIVATION], mark_object_used=False
+        )
     except ConfirmationKeyException:
         return render(request, "zerver/realm_reactivation_link_error.html")
     assert isinstance(realm, Realm)
     do_reactivate_realm(realm)
+    # TODO: After reactivating the realm, the confirmation link needs to be revoked in some way.
+
```

Comment:
Maybe we want an issue to track this actually? As it stands, a realm reactivation link can be freely re-used and never expires. Which isn't terrible, but definitely not desirable either - e.g. you can use it to reactivate it in the far future after you're no longer even the owner of the org, and the actual owner deactivated it. Or to reactivate it after forced deactivation of the realm by us due to abuse etc.

Reply:
Yeah, that sounds like the right way to fix this.